### PR TITLE
Add Google Analytics tracking

### PR DESCRIPTION
**Pull request by captainsafia**
Setting us up with tracking so we can get some cool stats.

Shamefully, this slows down launch time quite a bit.

Also, it asks the user whether or not they want to be tracked at every launch.

### CODE REVIEW

File: src/main/launch.js [hunk 1]
```diff
@@ -22,8 +22,26 @@ export function launch(filename) {
...
+    message: 'Would you like to enable Google Analytics?',
+    detail: 'We\'d like to enable tracking in this session of nteract. Why are ' +
+      'we doing this? Tracking usage helps us get statistics for grant funding, ' +
+      'the developement of localization features, and more. To read more about ' +
+      'how we use this information, please visit ' +
+      'https://github.com/nteract/nteract/blob/master/TRACKING.md.',
+  }, (choiceIndex) => {
+    let index;
+    if (choiceIndex === 0) {
+      index = path.join(__dirname, '..', '..', 'static', 'index-with-ga.html');
```
Comment (rgbkrk) [inline src/main/launch.js]: You can do all of the analytics loading as part of a dynamic epic or another way with the redux state tree, coordinating using `ipc` on launch. Then we don't have to duplicate all the code in the html below _and_ startup can be faster.

**Review by rgbkrk — CHANGES_REQUESTED** [review]

File: src/main/index.js [hunk 2]
```diff
@@ -201,3 +204,24 @@ fullAppReady$
       });
     });
   });
+
+browserWindowReady$
+  .skip(1)
```
Comment (rgbkrk) [inline src/main/index.js:209]: Does this fire more than once then?

**Comment by captainsafia** [comment]
> Does this fire more than once then?

Yeah, the skip is so that it doesn't run this on the initial splash window. But as before, it runs every time the user opens a new notebook.

I'm trying to see if there is some combination of operators wherein I can say "If this is the second time this event is triggered, show the dialog box and then load Google Analytics into the webpage. For all future emissions, just load GA into the webpage."

**Comment by rgbkrk** [comment]
> I'm trying to see if there is some combination of operators wherein I can say "If this is the second time this event is triggered, show the dialog box and then load Google Analytics into the webpage. For all future emissions, just load GA into the webpage."

I'd make this a global "configuration" thing, just like our theme setup - rely on `electron-json-storage` for now until we have a main thread config system. The config for this can likely be:

``` json
{
  "hasBeenPrompted": true,
  "send-metrics": true,
}
```

then again, you may only need the one field (if this wasn't set up, then we can assume a dialog):

``` json
{
  "send-metrics": true,
}
```

**Comment by captainsafia** [comment]
Closing in favor of another PR with the config approach.

**Pull request closed without merging**